src/main/kotlin/com/healthmetrix/labres/lab/LabController.kt

```diff
@@ -178,17 +288,47 @@ class LabController(
     fun ldtResult(
         @RequestHeader(HttpHeaders.AUTHORIZATION)
         labIdHeader: String,
+        @RequestParam(name = "issuer", required = false)
+        issuerId: String?,
         @RequestBody
         @Schema(description = "An LDT Document")
         ldtMessage: String
     ): ResponseEntity<UpdateStatusResponse> {
-        val labId = extractLabIdUseCase(labIdHeader)
-            ?: return UpdateStatusResponse.OrderNotFound.asEntity()
+        val labId = extractLabIdFrom(labIdHeader) ?: return UpdateStatusResponse.OrderNotFound.asEntity()
 
         val labResult = extractLdtResultUseCase(ldtMessage, labId)
             ?: return UpdateStatusResponse.InfoUnreadable.asEntity()
 
-        return updateResultUseCase(labResult).asEntity()
+        return updateResultUseCase(labResult).asUpdateStatusResponseEntity()
+    }
+
+    private fun extractLabIdFrom(labIdHeader: String): String? {
+        val (prefix, encoded) = labIdHeader.split(" ").also {
+            if (it.size != 2) return null
+        }
+
+        if (prefix != "Basic")
+            return null
+
+        val decoded = encoded.decodeBase64()
+            ?.split(":")
+            ?: return null
+
+        if (decoded.size != 2)
+            return null
+
+        return decoded.first()
+    }
```

Comment:
I pulled it in initially, cause to me it is deserialization / validation logic which is for me responsibility of the controller in oppose to a use case which for me should describe business logic.
After all, I found out, I will have to validate that the given labId has the permissions to update orders for a specified issuer or for an eon, to make sure, that e.g. lab A can not update orders by mistake that have been issued by another hospital / issuer.
So I will have to put some more logic to this and for me it would be the cleanest solution to implement this as a HTTP filter / preHandle interceptor which are components that can be injected for certain request paths by spring

Reply:
alright makes sense